Add AAD Auth Feature for Android (#1248)
microsoft/appcenter-sdk-android: CHANGELOG.md

# App Center SDK for Android Change Log

## Version 2.4.0 (Under development)

### App Center Auth

> * **[Feature]** Add support for Azure Active Directory (AAD) type authentication. Users with AAD apps on Azure can authenticate using one of three audiences viz. AzureADMyOrg (Single Tenant), AzureADMultipleOrgs (Multi Tenant) and AzureADandPersonalMicrosoftAccount (Multi Tenant and Personal Microsoft accounts).

Author: Still needs to be reviewed by the feature's PM.

Reviewer: Nice writing! It’s not immediately clear to me (and I’d imagine the user) what the terms viz, AzureADMyOrg, AzureADMultipleOrgs, AzureADandPersonalMicrosoftAccount means. What I’d recommend is to explain that explicitly in the language, and add a link for the users to learn more :)

Question: 
- is there a link that explains AAD app types single tenant, multi-tenant and personal MSA accounts? I could only find this, but that doesn’t capture personal MSA accounts https://docs.microsoft.com/en-us/azure/active-directory/develop/app-types#single-tenant-and-multi-tenant-apps. Would be good to include a link.
- what are the terms “azureADMyOrg” etc? Would be good if there are links to explain these terms as well. The general guidance of a good description is that it should be instantly clear to the layman user, with links to learn more if the terms are more complex.

Here’s a proposal, and if you have the links please link it below:
```suggestion
* **[Feature]** Add authentication support for Azure Active Directory (AAD) applications. Users can now connect an [AAD tenant and AAD single tenant (AzureADMyOrg), multi-tenant (AzureADMultipleOrgs) and multi-tenant and personal Microsoft account (AzureADandPersonalMicrosoftAccount)](https://www.xxx.com) applications.
```

> * **[Feature]** Add authentication support for Azure Active Directory (AAD) applications. Users can now connect an [AAD tenant and AAD single tenant (AzureADMyOrg), multi-tenant (AzureADMultipleOrgs) and multi-tenant and personal Microsoft account (AzureADandPersonalMicrosoftAccount)](https://docs.microsoft.com/en-us/azure/active-directory/develop/single-and-multi-tenant-apps) applications.

Reviewer: The link provided: https://docs.microsoft.com/en-us/azure/active-directory/develop/single-and-multi-tenant-apps, doesn't really match the terms used in the descriptions, and as a user, I still have questions such as "What is Azure ADMyOrg", "AzureADMultipleOrgs", "AzureADandPersonalMicrosoftAccount"? 

A suggestion would be to either 
1. Find a documentation link that directly explains what the terms "AzureADMyOrg", "AzureADMultipleOrgs", "AzureADandPersonalMicrosoftAccount" mean. Is there such a doc? If so, let's link it.
2. Directly explain the user scenarios from the [linked doc](https://docs.microsoft.com/en-us/azure/active-directory/develop/single-and-multi-tenant-apps). See below for an example of how it would look like. We don't have to use this doc, if there is better documentation that you'd recommend, let's use that.

```suggestion
* **[Feature]** Add authentication support for Azure Active Directory (AAD) applications. Users can now connect an AAD tenant with [AAD single-tenant or multi-tenant apps](https://docs.microsoft.com/azure/active-directory/develop/single-and-multi-tenant-apps) with the following audiences:
  - Accounts in this directory only
  - Accounts in any Azure AD directory
  - Accounts in any Azure AD directory and personal Microsoft accounts
```

fyi @manuchandel

### App Center Data

* **[Fix]** Reduced retries on Data-related operations to fail fast and avoid the perception of calls "hanging".

> * **[Feature]** Add support for Azure Active Directory (AAD) type authentication. Users with AAD apps on Azure can authenticate using one of three audiences viz. AzureADMyOrg (Single Tenant), AzureADMultipleOrgs (Multi Tenant) and AzureADandPersonalMicrosoftAccount (Multi Tenant and Personal Microsoft accounts).

Author: This is in Auth.

Reviewer 2: Made changes to go under Auth
